### PR TITLE
#9 Print day according to the index of the day

### PR DESCRIPTION
**Pull request by dipakthapaofficial**
a program that prints “Sunday”, “Monday”, ... “Saturday” if the int variable "day" is 0, 1, ..., 6, respectively. Otherwise, it shall print “Not a valid day”.

### CODE REVIEW

File: src/com/lftechnology/java/training/dipak/days/PrintDay.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
+package com.lftechnology.java.training.dipak.days;
+
+import java.util.Scanner;
+import java.util.logging.Logger;
+
+/*program called PrintDayInWord, which prints �Sunday�, �Monday�, ... �Saturday� if the int variable "day" is 0, 1, ..., 6, respectively.
+ * Otherwise, it shall print �Not a valid day�
+*/
+public class PrintDay {
```
Comment (bhuwang) [inline src/com/lftechnology/java/training/dipak/days/PrintDay.java]: @author missing, String printing is not proper due to copy paste effect :)

File: src/com/lftechnology/java/training/dipak/days/PrintDay.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	/**
+	 * <p>This method checks whether the day number matches to the day or not</p>
+	 * @param dayNumber
+	 * @return day
+	 * @author Dipak Thapa <dipakthapa@lftechnology.com>
+	 */
+	String getDay(int dayNumber){
+		String day;
+		switch(dayNumber){
+		case 1:	day="Sunday";
```
Comment (bhuwang) [inline src/com/lftechnology/java/training/dipak/days/PrintDay.java]: Consider maintaining CommonConstants.java file for hardcoded values.

File: src/com/lftechnology/java/training/dipak/days/PrintDay.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+*/
+public class PrintDay {
+	private final static Logger LOGGER = Logger.getLogger(PrintDay.class.getName());
+	/**
+	 * <p>This method checks whether the day number matches to the day or not</p>
+	 * @param dayNumber
+	 * @return day
+	 * @author Dipak Thapa <dipakthapa@lftechnology.com>
+	 */
+	String getDay(int dayNumber){
```
Comment (bhuwang) [inline src/com/lftechnology/java/training/dipak/days/PrintDay.java]: String getDay(int day) will be sufficient.

File: src/com/lftechnology/java/training/dipak/days/PrintDay.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+		case 6:day="Friday";
+				break;
+		case 7: day="Saturday";
+				break;
+		default: day="Not a valid day";
+				 break;
+		}
+		return day;
+	}
+
```
Comment (bhuwang) [inline src/com/lftechnology/java/training/dipak/days/PrintDay.java]: Make a habit of using Ctrl+Shit+F to beautify your code.

File: src/com/lftechnology/java/training/dipak/days/PrintDay.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+			}						
+			if(day<0){
+				break;
+			}
+			
+			PrintDay pd1=new PrintDay();
+			String displayText=pd1.getDay(day);
+			System.out.println(displayText);
+		}		
+		sc.close();
```
Comment (bhuwang) [inline src/com/lftechnology/java/training/dipak/days/PrintDay.java]: Again you are closing resources without using finally block or try-with-resource.

File: src/com/lftechnology/java/training/dipak/days/PrintDay.java [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+					} catch (NumberFormatException nfe) {
+						LOGGER.info("Characters entered.Re-enter the values.");
+					}
+				}
+				if (day <= 0) {
+					break;
+				}
+				PrintDay pd1 = new PrintDay();
+				String displayText = pd1.getDay(day);
+				System.out.println(displayText);
```
Comment (sanishmaharjan) [inline src/com/lftechnology/java/training/dipak/days/PrintDay.java]: Use logger to display day.

File: src/com/lftechnology/java/training/dipak/days/PrintDay.java [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+			}
+		} catch (Exception e) {
+			LOGGER.info("Exception occurred");
+		} finally {
+			sc.close();
```
Comment (sanishmaharjan) [inline src/com/lftechnology/java/training/dipak/days/PrintDay.java]: You can use try-with-resources feature to close the resources automatically.

**Comment by bhuwang** [comment]
Can one of the admins verify this patch?

**Pull request merged**